dask/array/tests/test_array_function.py

```diff
@@ -151,7 +151,7 @@ def test_unregistered_func(func):
     y = da.from_array(x, chunks=(50, 50))
 
     # Check if it's an equivalent array
-    assert_eq(x, y, check_meta=False)
+    assert_eq(x, y, check_meta=False, check_type=False)
```

Comment:
@jsignell I'm a little confused why this needs `check_type=False`; is the result here not an `EncapsulateNDArray`? Since the `EncapsulateNDArray` [implements `__array_function__`](https://github.com/dask/dask/blob/8800dc9dc062c55c88fa16cd5da8782a101161f5/dask/array/tests/test_dispatch.py#L54-L65), I'd think the `np.concatenate` in finalize would get dispatched to it.


I feel like it's important to check that this is indeed an `EncapsulateNDArray`; for example, if you called `compute` on a CuPy-backed array, you'd want to get a CuPy array out, not an ndarray where the data has been copied to host memory.

Reply:
I agree that it should not need a `check_type=False`